docs/src/main/sphinx/security/jwt-auth.rst

```diff
@@ -0,0 +1,158 @@
+`JWK <https://datatracker.ietf.org/doc/html/rfc7517>`_ format. These public
+keys are the counterpart of the private keys embedded in JSON web tokens.
+
+JWT authentication configuration
+--------------------------------
+
+Do not configure JWT authentication unless you are an administrator trying to
+secure Trino with a custom single sign-on or proxy solution. If that describes
+you, then enable JWT authentication by setting the :doc:`JWT authentication type
+<authentication-types>` in :ref:`etc/config.properties <config_properties>`,
+and specifying a URL or path to a key file:
+
+.. code-block:: properties
+
+    http-server.authentication.type=JWT
+    http-server.authentication.jwt.key-file=https://cluster.example.net/.well-known/jwks.json
+
+JWT authentication is typically used in addition to other authentication
+methods:
+
+.. code-block:: properties
+
+    http-server.authentication.type=PASSWORD,JWT
+    http-server.authentication.jwt.key-file=https://cluster.example.net/.well-known/jwks.json
+
+The following configuration properties are available:
+
+.. list-table:: Configuration properties for JWT authentication
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Property
+     - Description
+   * - ``http-server.authentication.jwt.key-file``
+     - Required. Specifies either the URL to a JWKS service or the path to a
+       PEM or HMAC file, as described below this table.
+   * - ``http-server.authentication.jwt.required-issuer``
+     - Optional. Specifies a string that must match the value of the JWT's
+       ``iss`` field to consider this JWT valid for this Trino server.
+       The ``iss`` field in the JWT identifies the principal that issued the
+       JWT.
+   * - ``http-server.authentication.jwt.required-audience``
+     - Optional. Specifies a string that must match the value of the JWT's
+       ``aud`` field to consider this JWT valid for this Trino server.
+       The ``aud`` field in the JWT identifies the recipients that the JWT is
+       intended for.
+   * - ``http-server.authentication.jwt.principal-field``
+     - Optional. String to identify the field in the JWT that identifies the
+       subject of the JWT. The default value is ``sub``.
+   * - ``http-server.authentication.jwt.user-mapping.pattern``
+     - Optional. A regular expression pattern to :doc:`map all user names
+       </security/user-mapping>` for this authentication system to the format
+       expected by the Trino server.
+   * - ``http-server.authentication.jwt.user-mapping.file``
+     - Optional. The path to a JSON file that contains a set of
+       :doc:`user mapping rules </security/user-mapping>` for this
+       authentication system.
+
+Use the ``http-server.authentication.jwt.key-file`` property to specify
+either:
+
+-  The URL to a JWKS endpoint service, where the URL begins with ``https://``
+   and ends with the port number. The JWKS service must be reachable from the
```

Comment:
I'm not sure the `and ends with the port number` is right.  I believe that anything that starts with http:// or https:// is considered JWKs

Reply:
Left port out.